src/Layers/xrRenderGL/glTexture.cpp

```diff
@@ -121,21 +121,31 @@ GLuint CRender::texture_load(LPCSTR fRName, u32& ret_msize, GLenum& ret_desc)
         glTexParameteri(target, GL_TEXTURE_BASE_LEVEL, 0);
         glTexParameteri(target, GL_TEXTURE_MAX_LEVEL, static_cast<GLint>(texture.levels() - 1));
 
-        if (gli::gl::EXTERNAL_RED != format.External) // skip for properly greyscale-alpfa fonts textures
+        if (gli::gl::EXTERNAL_RED != format.External) // skip for proper greyscale-alpha font textures
             glTexParameteriv(target, GL_TEXTURE_SWIZZLE_RGBA, &format.Swizzles[gli::SWIZZLE_RED]);
 
         glm::tvec3<GLsizei> const tex_extent(texture.extent());
 
+        GLenum err;
         switch (texture.target())
         {
         case gli::TARGET_2D:
         case gli::TARGET_CUBE:
             CHK_GL(glTexStorage2D(target, static_cast<GLint>(texture.levels()), format.Internal,
                            tex_extent.x, tex_extent.y));
+            if ((err = glGetError()) != GL_NO_ERROR)
+            {
+                Msg("Error: 0x%x: Invalid 2D texture: '%s'", err, fname);
+            }
             break;
         case gli::TARGET_3D:
+        case gli::TARGET_CUBE_ARRAY:
             CHK_GL(glTexStorage3D(target, static_cast<GLint>(texture.levels()), format.Internal,
                            tex_extent.x, tex_extent.y, tex_extent.z));
+            if ((err = glGetError()) != GL_NO_ERROR)
+            {
+                Msg("Error: 0x%x: Invalid 3D texture: '%s'", err, fname);
+            }
```

Comment:
Here's what I suggest:
```suggestion
            glTexStorage3D(target, static_cast<GLint>(texture.levels()), format.Internal,
                           tex_extent.x, tex_extent.y, tex_extent.z);
            const auto err = glGetError();
            if (err != GL_NO_ERROR)
            {
                VERIFY(err == GL_NO_ERROR, "Error when loading a texture");
                Msg("! OpenGL: 0x%x: Invalid 3D texture: '%s'", err, fname);
            }
```